### PR TITLE
Yor Action as Typescript Action

### LINKED ISSUE #5: Avoid docker action, convert to javascript
Docker actions have quite a few drawbacks:

- To quote github itself, _Because of the latency to build and retrieve the container, Docker container actions are slower than JavaScript actions._ From experience, we are not talking about milliseconds here, it's a lot slower, even if you are just pulling an existing image
- If the containers edit files (which yor-action does), it does it as the docker user. On a self hosted runner without rootless docker, it does it as root and then the agent is failing because it can't clean up those files.

The big benefit of docker actions are the ability to pack a whole environment and its dependencies. For yor it doesn't matter as its just a static binary.

I would be more than happy to submit a PR, I just want to check first if that's something the maintainers would be ok with.

Otherwise I will implement my own `setup-yor` action.

### PR DESCRIPTION
**Pull request by t-botz**
Fix #5 

# Benefits

1. Run Faster ( [see sample runs](https://github.com/t-botz/test-yor-action/actions/runs/1573552620) ). I have run it several times in different order, Docker basically more or less adds an overhead of 10s which IMO is huge.
2. I can now run it on my self hosted runners 😃 
3. No dependency on the `entrypoint.sh` script, better separation of concerns
4. No needs to update the yor version, the action reads it from github
5. User can pick specific yor version

It looks like a lot of code but not so much :) Most of it is coming from github reference repo and the dist folder is just the compiled Javascript.
I have preserved the current behaviour in order to be backward compatible even though I feel like some can be improved.

### CODE REVIEW

File: .github/workflows/yor-upgrade.yaml [hunk 1]
```diff
@@ -1,19 +0,0 @@
-name: yor-upgrade
```
Comment (t-botz) [inline .github/workflows/yor-upgrade.yaml:1]: not needed anymore, the code does it dynamically

**Review by nimrodkor — APPROVED** [review]
Thanks @t-botz !
Very very nice

File: .github/workflows/check-dist.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# `dist/index.js` is a special file in Actions.
+# When you reference an action with `uses:` in a workflow,
+# `index.js` is the code that will run.
+# For our project, we generate this file through a build process from other source files.
+# We need to make sure the checked-in `index.js` actually matches what we expect it to be.
+name: Check dist/
```
Comment (nimrodkor) [inline .github/workflows/check-dist.yml:6]: @t-botz The ending `/` is not necessary, github adds it by itself :)

File: README.md [hunk 3]
```diff
@@ -42,6 +42,7 @@ jobs:
       - name: Run yor action
         uses: bridgecrewio/yor-action@main
```
Comment (t-botz) [inline README.md:43]: Recommending to use `@main` is dangerous IMO and is why I felt like I could only be backward compatible.
Github guidelines are followed by most actions: https://docs.github.com/en/actions/creating-actions/about-custom-actions#using-release-management-for-actions

File: src/main.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    getArgs('--skip-dirs', 'skip_dirs'),
+    getArgs('--skip-resource-types', 'skip_resource_types'),
+    getArgs('--custom-tagging', 'custom_tags'),
+    getArgs('--output', 'output_format'),
+    getArgs('--config-file', 'config_file')
+  ].flat()
+
+  const logLevel = core.getInput('log_level')
+  if (logLevel) {
+    process.env['LOG_LEVEL'] = logLevel
```
Comment (t-botz) [inline src/main.ts:33]: This not idiomatic. IMO, that shouldnt be an input, the user should just provide the environment variable

File: src/main.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+function getArgs(flag: string, input: string): string[] {
+  const value = core.getInput(input)
+  if (value) return [flag, value]
+  return []
+}
+
+async function run(): Promise<void> {
+  const yorVersion = core.getInput('version')
+  const commitChanges = core.getInput('commit_changes') === 'YES'
```
Comment (t-botz) [inline src/main.ts:15]: Usually for boolean we just expect a more natural boolean, like `true`.
Github has `core.getBooleanInput`

File: .github/workflows/check-dist.yml (hunk 2 above)
Comment (t-botz) [reply .github/workflows/check-dist.yml:6]: Ha ha, This is a copy paste from github.
https://github.com/actions/typescript-action/blob/main/.github/workflows/check-dist.yml#L6

I can remove

**Comment by t-botz** [comment]
Woops sorry for the failed build, I have just fixed it

File: src/main.ts (hunk 4 above)
Comment (nimrodkor) [reply src/main.ts:33]: Agreed, let's remove it

File: src/main.ts (hunk 5 above)
Comment (nimrodkor) [reply src/main.ts:15]: Agreed, let's change it

**Comment by t-botz** [comment]
@nimrodkor Changes implemented! 😃

**Comment by t-botz** [comment]
@nimrodkor ping

**Pull request merged**